app/models/miq_alert_status/purging.rb

```diff
@@ -0,0 +1,21 @@
+class MiqAlertStatus < ApplicationRecord
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        ::Settings.miq_alert_status.history.keep_miq_alert_statuses.to_i_with_method.seconds.ago.utc
+      end
+
+      def purge_window_size
+        ::Settings.miq_alert_status.history.purge_window_size
+      end
+
+      def purge_scope(older_than)
+        where(:ems_id => nil)
```

Comment:
1. Are we still using `ems_id.nil?` Or are we using `deleted_at` for this?
2. You need to chain the 2 `where` clauses together.